docs/source/14_contribution/02_developer_contributor_guidelines.md

```diff
@@ -139,74 +139,48 @@ def count_truthy(elements: List[Any]) -> int:
-
-You need the dependencies from `test_requirements.txt` installed.
-
-```bash
-make test
-```
-
 ```eval_rst
 .. note::  We place `conftest.py <https://docs.pytest.org/en/latest/reference/fixtures.html>`_ files in some test directories to make fixtures reusable by any tests in that directory. If you need to see which test fixtures are available and where they come from, you can issue the following command ``pytest --fixtures path/to/the/test/location.py``.
 ```
 
-#### E2E tests (`behave`)
+### Hints on `pre-commit` usage
+[`pre-commit`](https://pre-commit.com) hooks run checks automatically on all the changed files on each commit but can be skipped with the `--no-verify` or `-n` flag:
 
 ```bash
-behave
+git commit --no-verify <...>
 ```
 
-#### Others
-
-Our CI / CD also checks that `kedro` installs cleanly on a fresh Python virtual environment, a task which depends on successfully building the documentation:
+All checks will run during CI build, so skipping checks on commit will not allow you to merge your code with failing checks. You can uninstall the `pre-commit` hooks by running:
 
 ```bash
-make build-docs
-```
-
-```eval_rst
-.. note::  This command will only work on Unix-like systems and requires ``pandoc`` to be installed.
+make uninstall-pre-commit
 ```
+`pre-commit` will still be used by `make lint`, but will not install the git hooks.
 
-### Hints on pre-commit usage
-
-The checks will automatically run on all the changed files on each commit.
-Even more extensive set of checks (including the heavy set of `pylint` checks)
-will run before the push.
-
-The pre-commit/pre-push checks can be omitted by running with `--no-verify` flag, as per below:
+### Developer Certificate of Origin
+We require that all contributions comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). This certifies that the contributor wrote or otherwise has the right to submit their contribution.
 
-```bash
-git commit --no-verify <...>
-git push --no-verify <...>
+All commits must be signed off by including a `Signed-off-by` line in the commit message:
 ```
-(`-n` alias works for `git commit`, but not for `git push`)
-
-All checks will run during CI build, so skipping checks on push will
-not allow you to merge your code with failing checks.
+This is my commit message
 
-You can uninstall the pre-commit hooks by running:
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
 
+The sign-off can be added automatically to your commit message using the `-s` option:
 ```bash
-make uninstall-pre-commit
+git commit -s -m "This is my commit message"
 ```
-`pre-commit` will still be used by `make lint`, but will not install the git hooks.
 
+If your PR is blocked due to unsigned commits then you will need to follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.
```

Comment:
```suggestion
To avoid needing to remember the `-s` flag, you might like to set up an [alias](https://git-scm.com/book/en/v2/Git-Basics-Git-Aliases) that automatically signs off any commits.

If your PR is blocked due to unsigned commits then you will need to follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.
```